packages/store/addon/-private/ts-interfaces/minimum-adapter-interface.ts

```diff
@@ -0,0 +1,500 @@
+type Dict<T> = import('./utils').Dict<T>;
+type Snapshot = import('../system/snapshot').default;
+type SnapshotRecordArray = import('../system/snapshot-record-array').default;
+type Store = import('../system/core-store').default;
+type ModelSchema = import('../ts-interfaces/ds-model').ModelSchema;
+type AdapterPopulatedRecordArray = import('../system/record-arrays/adapter-populated-record-array').default;
+type RelationshipSchema = import('./record-data-schemas').RelationshipSchema;
+
+type Group = Snapshot[];
+
+/**
+  ## Overview
+
+  In order to properly fetch and update data, EmberData
+  needs to understand how to connect to your API.
+
+  `Adapters` accept various kinds of requests from the store
+  and manage fulfillment of the request from your API.
+
+  ### Request Flow
+
+  When the store decides it needs to issue a request it uses the
+  following flow to manage the request and process the data.
+
+  - find the appropriate adapter
+  - issue the request to the adapter
+  - await the adapter's response
+    - if an error occurs reject with the error
+    - if no error
+      - if there is response data
+        - pass the response data to the appropriate serializer
+        - update the cache using the JSON:API formatted data from the serializer's response
+      - return the primary record(s) associated with the request
+
+
+  ### Implementing an Adapter
+
+  There are seven required adapter methods, one for each of
+  the primary request types that EmberData issues.
+
+  They are:
+
+  - findRecord
+  - findAll
+  - queryRecord
+  - query
+  - createRecord
+  - updateRecord
+  - deleteRecord
+
+  Each of these request types has a matching store method that triggers it
+  and matching `requestType` that is passed to the serializer's
+  `normalizeResponse` method.
+
+  If your app only reads data but never writes data, it is not necessary
+  to implement the methods for create, update, and delete.
+
+
+  data
+  normalizing data from the server API format into JSON:API, and
+  another for serializing records via `Snapshots` into the expected
+  server API format.
+
+  To implement a serializer, export a class that conforms to the structure
+  described by the [MinimumSerializerInterface](/ember-data/release/classes/MinimumSerializerInterface)
+  from the `app/serializers/` directory. An example is below.
+
+  ```ts
+  import EmberObject from '@ember/object';
+
+  async function fetchData(url, options = {}) {
+    let response = await fetch(`./${modelName}s/${id}`, options);
+    return response.toJSON();
+  }
+
+  export default class ApplicationAdapter extends EmberObject {
+    findRecord(_, { modelName }, id) {
+      return fetchData(`./${modelName}s/${id}`);
+    }
+  }
+  ```
+
+  ### Adapter Resolution
+
+  `store.adapterFor(name)` will lookup adapters defined in
+  `app/adapters/` and return an instance. If no adapter is found, an
+  error will be thrown.
```

Comment:
This seems misleading, because it doesn't account for the application adapter, which is described below

Reply:
How so?